base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -482,6 +482,19 @@ where B: BlockchainBackend
         fetch_orphan(&*db, hash)
     }
 
+    pub fn fetch_all_orphans(&self) -> Result<Vec<Block>, ChainStorageError> {
+        let db = self.db_read_access()?;
+        let mut result = vec![];
+        // TODO: this is a bit clumsy in order to safely handle the results. There should be a cleaner way
```

Comment:
I did some work that never got finished on lazy-loaded iterators (for LMDB but with the intention to do it in a db agnostic way) - which should replace the need for the for_each functions and having to allocate space for all items - and since `Result` implements `FromIterator`, errors can be handled cleanly.

Reply:
Yeah, this is a bit clumsy. I think this method will be temporary. Looking forward to using you new interface